lib/jwt-utils.js

```diff
@@ -364,6 +364,14 @@ function checkPayload(token, cb) {
       return cb(errors.INVALID_EXP());
     }
 
+    var clientExpired = iatParsed + config.expiration;
+
+    if (clientExpired < payload.exp) {
+      if (clientExpired < currentDate) {
+        return cb(errors.EXPIRED_JWT_BY_CLIENT(), token);
+      }
+    }
+
```

Comment:
All this is included in an if (payload.exp)... what if the incoming jwt does not have exp? we should calculate the expLimit in that case too.